src/scf/driver/scf_loop.cpp

```diff
@@ -107,32 +139,39 @@ MODULE_RUN(SCFLoop) {
     }
     auto e_nuclear = V_nn_mod.run_as<v_nn_pt>(qs_lhs, qs_rhs);
 
+    // Compute S
+    chemist::braket::BraKet s_mn(aos, simde::type::s_e_type{}, aos);
+    auto& s_mod   = submods.at("Overlap matrix builder");
+    const auto& S = s_mod.run_as<s_pt>(s_mn);
+
     wf_type psi_old = psi0;
     simde::type::tensor e_old;
-    const unsigned int max_iter = 3;
-    unsigned int iter           = 0;
 
-    while(iter < max_iter) {
-        // Step 2: Build old density
-        density_op_type rho_hat(psi_old.orbitals(), psi_old.occupations());
-        chemist::braket::BraKet P_mn(aos, rho_hat, aos);
-        const auto& P = density_mod.run_as<density_pt>(P_mn);
-        density_t rho_old(P, psi_old.orbitals());
+    density_op_type rho_hat(psi_old.orbitals(), psi_old.occupations());
+    chemist::braket::BraKet P_mn(aos, rho_hat, aos);
+    const auto& P = density_mod.run_as<density_pt>(P_mn);
+    density_t rho_old(P, psi_old.orbitals());
 
-        // Step 3: Old density is used to create the new Fock operator
+    const auto max_iter = inputs.at("max iterations").value<unsigned int>();
+    const auto e_tol    = inputs.at("energy tolerance").value<double>();
+    const auto g_tol    = inputs.at("gradient tolerance").value<double>();
+    unsigned int iter   = 0;
+
+    while(iter < max_iter) {
+        // Step 2: Old density is used to create the new Fock operator
         // TODO: Make easier to go from many-electron to one-electron
         // TODO: template fock_pt on Hamiltonian type and only pass H_elec
         const auto& f_new = fock_mod.run_as<fock_pt>(H, rho_old);
         const auto& F_new = Fock_mod.run_as<fock_pt>(H, rho_old);
 
-        // Step 4: New Fock operator is used to compute the new wavefunction
+        // Step 3: New Fock operator is used to compute the new wavefunction
         auto new_psi = update_mod.run_as<update_pt<wf_type>>(f_new, psi_old);
         const auto& new_cmos  = new_psi.orbitals();
         const auto& new_evals = new_cmos.diagonalized_matrix();
         const auto& new_c     = new_cmos.transform();
```

Comment:
```suggestion
        auto psi_new = update_mod.run_as<update_pt<wf_type>>(f_new, psi_old);
        const auto& cmos_new  = psi_new.orbitals();
        const auto& evals_new = cmos_new.diagonalized_matrix();
        const auto& c_new     = cmos_new.transform();
```
Everything else is named `X_new` or `X_old`, so these should be consistent